src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/createaccessrequirement/CreateOrUpdateAccessRequirementWizard.java

```diff
@@ -0,0 +1,103 @@
+package org.sagebionetworks.web.client.widget.accessrequirements.createaccessrequirement;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import org.sagebionetworks.repo.model.AccessRequirement;
+import org.sagebionetworks.repo.model.RestrictableObjectDescriptor;
+import org.sagebionetworks.web.client.context.SynapseReactClientFullContextPropsProvider;
+import org.sagebionetworks.web.client.jsinterop.CreateOrUpdateAccessRequirementWizardProps;
+import org.sagebionetworks.web.client.jsinterop.React;
+import org.sagebionetworks.web.client.jsinterop.ReactNode;
+import org.sagebionetworks.web.client.jsinterop.SRC;
+import org.sagebionetworks.web.client.widget.ReactComponentDiv;
+
+public class CreateOrUpdateAccessRequirementWizard implements IsWidget {
+
+  private final SynapseReactClientFullContextPropsProvider propsProvider;
+
+  private final ReactComponentDiv reactComponentDiv;
+  private RestrictableObjectDescriptor subject;
```

Comment:
Are these class member variables necessary (subject, accessRequriementId, onComplete, onCancel)?  I see them as being out of sync if this wizard view is reused (configure with subject, then later configured with a AR.  The subject would still be set).

Reply:
Good point!I included the class member variables so that the parent could call `setOpen` with only a boolean. However, I agree that currently the variables could be out of sync if the view was reused! I'll refactor to better handle that case.